### DISCUSSION TITLE
Table filter best practice - datepicker (Office hours: Oct 13, 2020)

### DISCUSSION THREAD

**Discussion by sang-yeo** [twilio-labs/paste#840] [Office hours] [answered]
Hello there, I'm working on Twilio Debugger page and like to know what's the best way to use a datepicker(console component) as a filter on Paste component. Added details in the signup form.

**Comment by richbachman** [comment, marked as answer]
Hi @sang-yeo. 

Thanks for posting this here. Is there anything in the research or mockups that can't be shared publicly? If not I'll add those links here, so the information is more visible. Thanks.

**Reply by sang-yeo** [reply]
Hey @richbachman, I added all necessary links into the meeting sign-up doc. I wasn't sure the research data link I posted in the sign up sheet can be shared  publicly.

**Reply by richbachman** [reply]
@sang-yeo would you mind adding a little more information about what you're looking for? More specific questions would help us work-through any answers/suggestions before office hours. Thanks.

**Reply by sang-yeo** [reply]
Sure, here are list of questions I'd like to get answers from you all. 

- I have two design concepts regarding table filter. one is filter as a part of table and the other one is filter in the disclosure component. Which interaction will help our customers better? 
- When will be the date picker component available? 
- If the component is not available within 2 month, what will be the best practice using the console date picker component? 
- Can I use progressive disclosure interaction in the disclosure component?

**Reply by serifluous** [reply]
I'll leave some more comments in Abstract, but here's some quick answers on these questions:

> * I have two design concepts regarding table filter. one is filter as a part of table and the other one is filter in the disclosure component. Which interaction will help our customers better?

Filters don't need to be in disclosures, but it could be helpful in some cases depending on the customer use case. For customers going to Debugger, would they need to minimize the filters? Is that a major use case, or would that extra UI be distracting?

> * When will be the date picker component available? If the component is not available within 2 month, what will be the best practice using the console date picker component?

We're aiming for an alpha release in January. If you need it sooner, let's chat during office hours!

> * Can I use progressive disclosure interaction in the disclosure component?

This seems fine to me since the date range fields would appear after all the other form fields, but I need to double-check to make sure.

**Reply by sang-yeo** [reply]
Here's another [concept](https://drive.google.com/file/d/16uV1_LD1V4ASXvN1eoOqfYJaPxvtEjvb/view?usp=sharing) that I'd like to get Dsys feedback on.

**Reply by SiTaggart** [reply]
Hi @sang-yeo. 

What are the exact questions you are looking for answers on? What are the features you need feedback on? Whats type of feedback are you looking for?

If they are unrelated to table filters, please start a new discussion surrounding the type of feedback on the new feature you require.

Please take the time to make them as detailed and as specific as possible.

**Reply by sang-yeo** [reply]
Cool. I applied the feedback I've got from the office hour to the concept. I'd like to get feedback on...  

1. Overall interaction using the new datepicker
2. Do we need to put background color on the new table header? 
3. Table footer treatment for dynamic search result - we have a background color on every other row how does this interact with footer when colored row is above the table footer?

**Reply by serifluous** [reply]
> Overall interaction using the new datepicker

Overall, this seems fine to me. I wonder if, even if a customer selects a preset value on the "Date range field" like "Last 30 days", a customer might still want to see the date-time picker to fine-tune the time range selection? Just a thought—you'll probably get a better sense of this from testing.

> Do we need to put background color on the new table header?

Nope, not yet. Follow the component style you see on the doc site for now. Once we make design updates to it, we'll give it to you through the symbol and your engineering team will get it through component updates.

> Table footer treatment for dynamic search result - we have a background color on every other row how does this interact with footer when colored row is above the table footer?

Can you say a bit more about what you need to put in a Table Footer? Is it just pagination?

--------------

Other questions/feedback:
1. One thing that's still throwing me off a little is the "Group by" form field. Is it filtering or sorting? Does it affect both the Table and the Graph?
2. For the "Alert type" form field, have you tried this using checkboxes? Since you're allowing customers to select 1 of 2 alert types, or both alert types, checkboxes could make that choice a little clearer.
3. On the Request Inspector modal, I'd recommend moving the "Go to event details page" link into the Modal Body. Footer actions usually affect interactions on the modal itself, like moving forward or backward or closing the modal.
4. On the Request Inspector, would it make sense to change the "Request" and "Response" disclosures to headings, so that you can avoid nesting disclosures?

**Reply by sang-yeo** [reply]
Thank you @serifluous 
> Can you say a bit more about what you need to put in a Table Footer? Is it just pagination?<

Kind of... it's about footer background color, I don't see any background color info on the Paste doc so I assume no color for the table footer correct? 

> One thing that's still throwing me off a little is the "Group by" form field. Is it filtering or sorting? Does it affect both the Table and the Graph?<

It's a sorting and it will affect both. I've been updating the concept. You can see it [here](https://share.goabstract.com/bf56d29b-c928-4c50-b5f8-e6bca8724a75?collectionLayerId=b8f8dd7e-beef-4efa-b3e8-f0e26f64dd25&mode=design) 

> For the "Alert type" form field, have you tried this using checkboxes? Since you're allowing customers to select 1 of 2 alert types, or both alert types, checkboxes could make that choice a little clearer<

Current Debugger uses the checkbox option and it allows user to unselect both types which is null

> On the Request Inspector, would it make sense to change the "Request" and "Response" disclosures to headings, so that you can avoid nesting disclosures?<

Could you tell me more? If I make the "request" and "response to headings but keep the rest of them in multiple disclosures or one?

**Reply by serifluous** [reply]
> Kind of... it's about footer background color, I don't see any background color info on the Paste doc so I assume no color for the table footer correct?

Correct, no color for Table Footer right now. However, if you need Pagination, we're going to be kicking off work for that soon and can give you some better guidance for where that should be placed and how that should look in a few weeks. 

> Current Debugger uses the checkbox option and it allows user to unselect both types which is null

👍 Filter groups are on our roadmap, so this use case is good to know for us.

> Could you tell me more? If I make the "request" and "response to headings but keep the rest of them in multiple disclosures or one?

Yep, keep the rest of them in multiple disclosures.
